src/partial-assembly.md

```diff
@@ -0,0 +1,165 @@
+# Performance and Partial Assembly
+
+This document provides a brief overview of the tensor-based high-performance
+and partial assembly features in MFEM. In the traditional finite element
+setting, the operator is assembled in the form of a matrix. The action of the
+operator is computed by multiplying with this matrix. At high orders this
+requires both a large amount of memory to store the matrix, as well as many
+floating point operations to compute and apply it. _Partial assembly_ is a
+technique that allows for efficiently applying the action of finite element
+operators efficiently without forming the corresponding matrix. This is
+particularly important when running on GPUs.
+
+Partial assembly is enabled at the level of the `BilinearForm` by setting
+the assembly level:
+```c++
+a->SetAssemblyLevel(AssemblyLevel::PARTIAL);
+```
+Once partial assembly is enabled, subsequent calls to functions such as
+`FormLinearSystem` will result in an `Operator` that represents the action of
+the bilinear form `a`, without assembling a matrix. This functionality is
+illustrated in several [MFEM examples](examples.md), including example 1, 3, 6,
+9, 24, and 26.
+
+## Preconditioning with Partial Assembly
+
+When using partial assembly, the system matrix is no longer available for
+constructing preconditioners. This means that some of the standard
+preconditioners in MFEM such as `HypreBoomerAMG` and `GSSmoother` cannot be
+used.
+
+MFEM allows for the efficient construction of diagonal (Jacobi) smoothers for
+partially assembled operators on quad and hex meshes using the class
+`OperatorJacobiSmoother`. This class efficiently assembles the diagonal of the
+corresponding matrix, exploiting the tensor-product structure for efficient
+evaluation.
+
+MFEM also allows for Chebyshev smoothing with partial assembly using the class
```

Comment:
If possible, it would be helpful to have a link to some external reference describing Chebyshev (and perhaps another to Jacobi) smoothing.